src/main/java/de/zalando/zmon/dataservice/data/KairosDBStore.java

```diff
@@ -88,13 +91,14 @@ public static HttpClient getHttpClient(int socketTimeout, int timeout, int maxCo
     }
 
     @Autowired
-    public KairosDBStore(DataServiceConfigProperties config, DataServiceMetrics metrics) {
+    public KairosDBStore(DataServiceConfigProperties config, DataServiceMetrics metrics, RedisDataPointsStore redisDataPointsStore) {
         this.metrics = metrics;
         this.config = config;
+        this.redisStore = redisDataPointsStore;
         this.resultSizeWarning = config.getResultSizeWarning();
 
 
-        if (config.isKairosdbEnabled()) {
+        if (config.isKairosdbEnabled() && !config.isDatapointsRedisEnabled()) {
```

Comment:
Let's put writing to our consumer, be that by writing directly to Kairos or posting to another service, behind an interface instead of using a flag.